photutils/psf.py

```diff
@@ -808,11 +821,28 @@ def psf_photometry(data, positions, psf, fitshape=None,
                 fit_infos.append(fitter.fit_info)
             if fit_messages is not None:
                 fit_messages.append(fitter.fit_info['message'])
+            if param_uncert:
+                if fitter.fit_info['param_cov'] is not None:
+                    uncert.append(np.sqrt(np.diag(\
+                            fitter.fit_info['param_cov'])))
+                else:
+                    warnings.warn("uncertainties on fitted parameters " +
+                                  "cannot be computed because the fit may " +
+                                  "be unsuccessful", AstropyUserWarning)
+                    uncert.append((None, None, None))
     else:
         raise ValueError('Invalid photometry mode.')
 
     if fit_messages is not None:
         result_tab['fit_messages'] = fit_messages
+    if param_uncert:
+        uncert = np.array(uncert)
+        result_tab.add_column(Column(name='flux_uncertainty', unit=fluxunit,
+                                     data=uncert[:, 0]))
+        result_tab.add_column(Column(name='x_uncertainty', unit=None,
+                                     data=uncert[:, 1]))
+        result_tab.add_column(Column(name='y_uncertainty', unit=None,
+                                     data=uncert[:, 2]))
```

Comment:
The way this is done right now, you only ever get uncertainties on `flux`/`x`/`y`.  But the *fitting* will sometimes use other parameters (and in fact might even not do  `flux`/`x`/`y`if the fitter has them set to "fixed").  So I suggest adapting this to instead accept all of the "non-fixed" parameters.  To see how that works, check out the loop starting on this line : https://github.com/astropy/photutils/blob/2ec0a32dc0990a5ada705a323d04520c9c4fc883/photutils/psf.py#L762

You can see there that it does basically just what these three lines do, but in a way that is more general for other non-fixed parameters.